net/api/client/client.go

```diff
@@ -59,153 +57,206 @@ func (c *Client) GetHostID(ctx context.Context) (peer.ID, error) {
 	return peer.IDFromBytes(resp.PeerID)
 }
 
-func (c *Client) CreateThread(ctx context.Context, creds thread.Credentials, opts ...core.KeyOption) (info thread.Info, err error) {
-	signed, err := signCreds(creds)
+func (c *Client) GetToken(ctx context.Context, identity thread.Identity) (tok thread.Token, err error) {
```

Comment:
Here's a client requesting a token with an `Identity`. 
1. Send the remote a pubkey
2. Wait for a challenge
3. Sign the challenge and send back the signature
4. Wait for the final token